common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -55,6 +91,43 @@ trait Logging {
     log_
   }
 
+  implicit class LogStringContext(val sc: StringContext) {
+    def log(args: Any*): MessageWithContext = {
+      val processedParts = sc.parts.iterator
+      val sb = new StringBuilder(processedParts.next())
+      lazy val map = new java.util.HashMap[String, String]()
+
+      args.foreach { arg =>
+        arg match {
+          case mdc: MDC =>
+            sb.append(mdc.value)
+            if (Logging.isStructuredLoggingEnabled) {
+              map.put(mdc.key.toString.toLowerCase(Locale.ROOT), mdc.value)
+            }
+          case other =>
+            // Note: all the arguments are supposed to be MDCs, but we only throw an exception
+            //       if we are running in test mode. This is to avoid breaking existing code.
```

Comment:
Shall we change `args: Any*` to `args: MDC*` ? So we can avoid the match path.

Reply:
Thanks, this is a great suggestion!